### PR TITLE
[DDW-414] Bump electron and related packages

### PR DESCRIPTION
**Pull request by yakovkaravelov**
This PR updated electron and related packages.
[Jira Ticket](https://jira.iohk.io/browse/DDW-414)

**Latest test document**
[Regression Spreadsheet](https://docs.google.com/spreadsheets/d/1ZEvSxJWwwNQA5Grqs_b6Wh3qkt0sfijGNsOwYQSFHfI/edit?usp=sharing)

**Previous test documents**
[Regression Spreadsheet 2](https://docs.google.com/spreadsheets/d/1ky065m3LZKnqL8PgKuLzmWWKCYnOy2iXBgcl4KQ3u38/edit?usp=sharing)
[Regression Spreadsheet 1](https://docs.google.com/spreadsheets/d/1PUtK8AR9DTnklfHPNhDlCAt3eKkLsJ7wLChiytActQI/edit?usp=sharing)
[HW Regression Spreadsheet](https://docs.google.com/spreadsheets/d/1wWa04fcafc5kmGKxHFViTYzzbtSAlXQB-7qbo_231Ok/edit?usp=sharing)
OLD test document see [here](https://docs.google.com/spreadsheets/d/14r6z_m58VQkzWM6TWE1l41A5OjxOVjoxeCtLcdK2Rck/edit#gid=287361067)

## Todos
- [x] Update electron and related packages
- [x] Update nix shell scripts
- [x] Revert this commit before merging https://github.com/input-output-hk/daedalus/pull/2206/commits/e49537bf8b5daabe909ba9506f257c1162c4744b -> completed in https://github.com/input-output-hk/daedalus/pull/2206/commits/cc71e3403b95522d8d20a3b1bcc8730d2154c0d0

## Testing Checklist
- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1602727073099700)

### App
- [x] Make sure app is working properly

### Storybook
- [x] Make sure all the stories are working properly

## Testing Summary

QA Summary for build 18923:

QA Summary for build 18923:

New
- [macOS] The error "transaction confirmation failed" is displayed if you create a transaction for Ledger S and connect the device after. Check [this](https://drive.google.com/file/d/1AY-sUIDrzKap88KuPp78N2oquTNdJJtY/view?usp=sharing) screenshot.

- [Windows] Pair a Ledger S device, wait for it to sync and unlock it. Create a transaction without connecting the device, you'll see that Daedalus asks to "Run cardano app" instead of "Connect the Ledger S". A lot of errors are visible in the console. Check [this](https://drive.google.com/file/d/13ENOwBgwAofb82fq-4y42YQoAsICve9O/view?usp=sharing) screenshot.
Note: If you connect the device at this stage, you still can confirm the transaction.

- [Intermittent - Windows] When trying to delete a Ledger S wallet, an error on the console is displayed. The process of deletion can take a while and if you restart Daedalus, the wallet is still there. Check [this](https://drive.google.com/file/d/1-Z2WYHAMrb44xRw_WNr4mggGvIn4ghLp/view?usp=sharing) screenshot to see the error.

Pending
- I attempted to send ADA with balance enough to cover transaction fee. But at the last stage, after clicking "Send" button, the error displayed was "Not enough ada for fees...". Please see [fee1,](https://drive.google.com/file/d/1WGg5l-BCuBdhyKqhdubCPbuQ-KinWgpQ/view?usp=sharing) [fee2](https://drive.google.com/file/d/1U-1LgcGdhz71zACEvLngDuoLYymEx9Wc/view?usp=sharing), [fee3](https://drive.google.com/file/d/1ZuuqCDmlmqgolILK5TlJ9ygZ60r5C6l3/view?usp=sharing)

Fixed
- I was able to pair Nano S with Mac Catalina, but wasn't able to execute any transaction. On Daedalus Transaction screen displayed "Export the public key on your device" but Nano S device displayed "Waiting for commands"

- Daedalus crashed when I attempted to send ADA but the balance wasn't enough to cover transaction fee. I managed to duplicate the issue with other wallets too (both Byron and Shelley). Please see [video](https://drive.google.com/file/d/1GT97YRh_P5RmV3UuRgG3Qrto2AEtAaop/view?usp=sharing)

- The "Blank screen fix" issue is ok now, but the state is not persisted after closing Daedalus. To clarify: In this build If you choose "Blank screen fix", when you close and reopen Daedalus, it is off again. In the version from production, it's a state that doesn't change until you deactivate it.

- After "Blank screen fix" is active, it can't be deactivated. Check [this](https://drive.google.com/file/d/1sbCyJDrtvTZpJcPM--agUByxnFG92J95/view?usp=sharing) video. Happens on all platforms. Reported [here](https://github.com/input-output-hk/daedalus/pull/2206#issuecomment-882774936)

- Error on BigSur when closing Daedalus. Check [screenshot](https://drive.google.com/file/d/1NAvCCzMC6tvBq25H83yxkqUXTKbZ5eA2/view?usp=sharing). (Reported via Slack)

- Each time I close Daedalus it reopens by itself. Happens on all platforms. Check [this](https://drive.google.com/file/d/1yl5qeuSwIMbwhLBHc81aYJWtcvr2Vk8l/view?usp=sharing) video. Reported [here](https://github.com/input-output-hk/daedalus/pull/2206#issuecomment-864016992).

- Daedalus restarts after choosing "Blank screen fix", but the fix is disabled (it should look like [this](https://drive.google.com/file/d/1nOaDXqQ2qsZJhX8cIpslm_KD86zGnULX/view?usp=sharing)). Check [this](https://drive.google.com/file/d/1wb3QmWzZYhF2vvzqMwz_GQoa22bJbJ-D/view?usp=sharing) video. Reported [here](https://github.com/input-output-hk/daedalus/pull/2206#issuecomment-864016992).

- Can't pair Ledger devices on macOS and Windows. Reported [here](https://github.com/input-output-hk/daedalus/pull/2206#issuecomment-863894377), [here](https://github.com/input-output-hk/daedalus/pull/2206#issuecomment-863907769), [here](https://github.com/input-output-hk/daedalus/pull/2206#issuecomment-863962152) and [here](https://github.com/input-output-hk/daedalus/pull/2206#issuecomment-864016992).

- Daedalus doesn't restart after selecting "Blank screen fix". Reported [here](https://github.com/input-output-hk/daedalus/pull/2206#issuecomment-863527646).

- When you select "Blank screen fix", the Cardano node stops, but Daedalus doesn't restart. If you close it manually after this, it would restart, but the "Blank screen fix" won't be enabled. Check [this](https://drive.google.com/file/d/1FDrT7c8yEouqXNAIJrTPklS3qgC7uZHR/view?usp=sharing) video. Reported [here](https://github.com/input-output-hk/daedalus/pull/2206#issuecomment-874193215).

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [x] There are no *flow* errors or warnings (`yarn flow:test`)
- [x] There are no *lint* errors or warnings (`yarn lint`)
- [x] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [x] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [x] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

**Comment by ManusMcCole** [comment]
Hi @yakovkaravelov. When attempting to open the state directory path in Daedalus Diagnostics Screen the Open button would not work and I got this error in the console. See <a href="https://drive.google.com/file/d/1IT1iiz7nr_N--2rRXmocp6ua64iQjVpA/view?usp=sharing">here</a>

**Comment by ManusMcCole** [comment]
Hi @yakovkaravelov. I added two more items to my comment above and I updated the test summary in the PR Description

**Comment by ManusMcCole** [comment]
Hi @yakovkaravelov . Here is the QA Summary for build 14590.

**Still Pending** Issues. I was able to reproduce the 2 issues below on the new build

1 - There is an error in the console "State snapshot log file creation failed". This is caused by downloading logs. The logs downloaded succesfully success message is still shown. See <a href="https://drive.google.com/file/d/1wdBNuEeR_eNdRgLeXi7g8u33FzYZG02F/view?usp=sharing">here</a>

2 - Not able to run Daedalus on Linux. Attempted with both MainNet and TestNet. The only log available is this [see log](https://drive.google.com/file/d/1f7muVgzwwSc8ZKWZvT72HezsqdGRwbdt/view?usp=sharing)


 *Fixed *
1 - When attempting to open the state directory path on Windows in Daedalus Diagnostics Screen the Open button would not work and I got this electron error in the console. See <a href="https://drive.google.com/file/d/1IT1iiz7nr_N--2rRXmocp6ua64iQjVpA/view?usp=sharing">here</a>
Same issue on [Mac](https://drive.google.com/file/d/1VBJLBK2E5V5-333IQXCo3Wy0zWpnDBPK/view?usp=sharing)

2 - When attempting to send a transaction that is not possible because I do not have enough funds an incorrect error message "Cannot calculate fees while there are pending transactions" is displayed. There are no pending transactions in the wallet. See <a href="https://drive.google.com/file/d/1WHGOlu1FlQY_sj-Sj2JnudL731WrfDzy/view?usp=sharing">here</a> 

**Test Document**
For test document see [here](https://docs.google.com/spreadsheets/d/14r6z_m58VQkzWM6TWE1l41A5OjxOVjoxeCtLcdK2Rck/edit#gid=287361067)

**Comment by nikolaglumac** [comment]
> 1 - On Mac and Windows.There is an error in the console "State snapshot log file creation failed". This is caused by downloading logs. The logs downloaded succesfully success message is still shown. See here

@ManusMcCole does this happen on all networks (Mainnet, Testnet, Shelley QA, Flight...)?

cc @yakovkaravelov

**Comment by ManusMcCole** [comment]
Hi @yakovkaravelov and @nikolaglumac. For "1 - On Mac and Windows.There is an error in the console "State snapshot log file creation failed". This is caused by downloading logs. The logs downloaded succesfully success message is still shown. See <a href="https://drive.google.com/file/d/1wdBNuEeR_eNdRgLeXi7g8u33FzYZG02F/view?usp=sharing">here</a>" this issue happens on all Networks on Windows and Mac. I cannot test Linux as it will not launch

Also for "2 - When attempting to send a transaction that is not possible because I do not have enough funds an incorrect error message "Cannot calculate fees while there are pending transactions" is displayed. There are no pending transactions in the wallet. See <a href="https://drive.google.com/file/d/1WHGOlu1FlQY_sj-Sj2JnudL731WrfDzy/view?usp=sharing">here</a>" it was marked as fixed but today now I notice sometimes there is a delay where Daedalus takes a good few seconds to realise the user does not have enough UTXO for the transaction. The message calculating fees seems to have higher precedence over the "not enough ada.Try sending a smaller amount". See <a href="https://drive.google.com/file/d/1JWdAqnSoawu6Vqjx3p2knpx0-IO2K9WW/view?usp=sharing">video from newest build</a>. Im adding this to the report now

**Comment by nikolaglumac** [comment]
> Also for "2 - When attempting to send a transaction that is not possible because I do not have enough funds an incorrect error message "Cannot calculate fees while there are pending transactions" is displayed. There are no pending transactions in the wallet. See here" it was marked as fixed but today now I notice sometimes there is a delay where Daedalus takes a good few seconds to realise the user does not have enough UTXO for the transaction. The message calculating fees seems to have higher precedence over the "not enough ada.Try sending a smaller amount". See video from newest build. Im adding this to the report now

@ManusMcCole I seriously doubt this is specific to this PR. Please test on other builds (2.4.0 release or develop builds) before adding this to the list of issues of this PR.

**Comment by mioriohk** [comment]
Hi @yakovkaravelov . I have retested on build 15000 on all 3 platforms. Could replicate all the pending issues as listed in "Testing Summary" please see [here](https://github.com/input-output-hk/daedalus/pull/2206#issue-503751345)

**Comment by yakovkaravelov** [comment]
The first issue was fixed.
So the following issue is only left.
```Not able to run Daedalus on Linux. Attempted with both MainNet and TestNet.```
~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Installing `daedalus-2.4.0-testnet-15056.bin` worked.
But after installing, running `daedalus-testnet` generated error as following.
![image](https://user-images.githubusercontent.com/6207747/98214609-8f9d3000-1efb-11eb-8feb-e00cdbfa1bfe.png)

* Couldn't figure out what `The updater file is missing!` error message means.
* Based on error message `/bin/electron: error while loading shared libraries: libdrm.so.2: cannot open shared object file: No such file or directory`, tried to install missing dependencies.
Referenced this: https://github.com/xorilog/docker-telegram/pull/13/files
But those packages were already installed.

I suspect there is some error in `default.nix`, `linux.nix`, `shell.nix` or `yarn2nix.nix`.

**Comment by nikolaglumac** [comment]
> The first issue was fixed.
> So the following issue is only left.
> `Not able to run Daedalus on Linux. Attempted with both MainNet and TestNet.`
> 
> ```
> Installing `daedalus-2.4.0-testnet-15056.bin` worked.
> But after installing, running `daedalus-testnet` generated error as following.
> image
> 
> * Couldn't figure out what `The updater file is missing!` error message means.
> * Based on error message `/bin/electron: error while loading shared libraries: libdrm.so.2: cannot open shared object file: No such file or directory`, tried to install missing dependencies.
> Referenced this: https://github.com/xorilog/docker-telegram/pull/13/files
> But those packages were already installed.
> 
> I suspect there is some error in `default.nix`, `linux.nix`, `shell.nix` or `yarn2nix.nix`.
> ```

@cleverca22 @disassembler can you guys please help us with this?

**Comment by yakovkaravelov** [comment]
@cleverca22 @disassembler Need some help for this.

**Comment by nikolaglumac** [comment]
@disassembler @cleverca22 this one has to be dealt with in February. We count on your support.

**Review by miorsufianiohk — CHANGES_REQUESTED** [review]
Installation error observed when attempting to install build 18347 on Mac Testnet. See [screenshot](https://drive.google.com/file/d/1A_o2gLnu-nyN7_-cCYIWbs5OzS-kJkX8/view?usp=sharing)

**Comment by gabriela-ponce** [comment]
@danielmain On build 18349, Daedalus doesn't restart after selecting "Blank screen fix".
Check [this](https://drive.google.com/file/d/1EIaHLFfGe_k73uphw1Plzfq8zbC1n_Qj/view?usp=sharing) video for the current build (I didn't record it, but there were no errors in the console). [This](https://drive.google.com/file/d/1UTxpfj8D396J-rSDlCjsL8MBkAeyUyqA/view?usp=sharing) is how it works on the last build on develop branch. 
It reproduces in Ubuntu 20.04 and Windows 10.

**Comment by ManusMcCole** [comment]
Hi @danielmain. Ledger Nano S and Ledger X are not connecting to Daedalus on Mac. The “Connect your device and enter…..” message remained after the device is physically connected and unlocked. No errors on the console Daedalus just never recognises the device. Even after many attempts will not connect even once. Connected succcesfully on Linux. I have tested on Mac I will check now on Windows to see if the errors exists there too

**Comment by ManusMcCole** [comment]
Hi @danielmain. It would not connect either on windows. this time an error did appear on the [console](https://drive.google.com/file/d/1P7IPBesSqwk5FcDIIgoN0gdNRU3gpMDU/view?usp=sharing). Daedalus would recognise device and ask to launch cardano application on your device but does not proceed after the app is launched

**Comment by danielmain** [comment]
> @danielmain On build 18349, Daedalus doesn't restart after selecting "Blank screen fix".
> Check [this](https://drive.google.com/file/d/1EIaHLFfGe_k73uphw1Plzfq8zbC1n_Qj/view?usp=sharing) video for the current build (I didn't record it, but there were no errors in the console). [This](https://drive.google.com/file/d/1UTxpfj8D396J-rSDlCjsL8MBkAeyUyqA/view?usp=sharing) is how it works on the last build on develop branch.
> It reproduces in Ubuntu 20.04 and Windows 10.

Hi @gabriela-ponce thank you for your time testing this PR. The issue is now fixed in Build #18358. You should now be able to restart Daedalus with "Blank screen fix". See [Video](https://drive.google.com/file/d/11p3ck4YZ5P7Q3hfK7EHnIT8ee_B8k30k/view?usp=sharing):

**Comment by ManusMcCole** [comment]
Hi @danielmain unfortunately the same issues have arisen again even after i upgraded to Ledger firmware 2.0.0 and also Cardano App 2.3.2 [Firmware status](https://drive.google.com/file/d/1s2Y0K5P7qBSl9XiNHFPM-bOXmyeC6DLE/view?usp=sharing). I Performed a clean install of Daedalus and state directory was deleted

Same error on windows as before here is the new [screenshot](https://drive.google.com/file/d/1-HyzG1l5iZrjJYa4vt4TXWAzHho8pG08/view?usp=sharing)

On Mac it is once again The “Connect your device and enter…..” message remains after the device is physically connected and unlocked.

**Comment by gabriela-ponce** [comment]
@danielmain on the build 18358:

- Each time I close Daedalus it reopens by itself. Happens on all platforms. Check [this](https://drive.google.com/file/d/1yl5qeuSwIMbwhLBHc81aYJWtcvr2Vk8l/view?usp=sharing) video.
- Daedalus restarts after choosing "Blank screen fix", but the fix is disabled (it should look like [this](https://drive.google.com/file/d/1nOaDXqQ2qsZJhX8cIpslm_KD86zGnULX/view?usp=sharing)). Check [this](https://drive.google.com/file/d/1wb3QmWzZYhF2vvzqMwz_GQoa22bJbJ-D/view?usp=sharing) video.
- I was able to reproduce the same issues that @ManusMcCole reported for Ledger S on macOS and Windows using firmware version 2.0.0 and Cardano app 2.3.2.

**Comment by gabriela-ponce** [comment]
@danielmain I checked the two first issues from [this](https://github.com/input-output-hk/daedalus/pull/2206#issuecomment-864016992) comment on the latest build (18473). Here are my findings:

- When you select "Blank screen fix", the Cardano node stops, but Daedalus doesn't restart. If you close it manually after this, it would restart, but the "Blank screen fix" won't be enabled. Check [this](https://drive.google.com/file/d/1FDrT7c8yEouqXNAIJrTPklS3qgC7uZHR/view?usp=sharing) video.
This reproduces on all platforms. We'll continue the regression and updating [this](https://docs.google.com/spreadsheets/d/1PUtK8AR9DTnklfHPNhDlCAt3eKkLsJ7wLChiytActQI/edit?usp=sharing) document. We'll let you know if there are other findings on the latest build.

**Comment by nikolaglumac** [comment]
@danielmain let's make sure to revert this commit before merging: https://github.com/input-output-hk/daedalus/pull/2206/commits/e49537bf8b5daabe909ba9506f257c1162c4744b

**Comment by gabriela-ponce** [comment]
@danielmain On build 18620 I can reproduce the following issue:

- After "Blank screen fix" is active, it can't be deactivated. Check [this](https://drive.google.com/file/d/1sbCyJDrtvTZpJcPM--agUByxnFG92J95/view?usp=sharing) video. Happens on all platforms.

**Comment by danielmain** [comment]
After a deep research by debugging all affected products, I can summarize what is the call stack that produces the crash after closing Daedalus.

The problem start as follows:

### daedalus
- source/main/index.js:117
- source/main/windows/main.js:68
- source/main/ipc/index.js:50
- source/main/ipc/getHardwareWalletChannel.js:443 => **TransportNodeHid.listen(observer)**

### hw-transport-node-hid
- packages/hw-transport-node-hid/src/TransportNodeHid.ts:79 => listenDevices(
- packages/hw-transport-node-hid/src/listenDevices.ts:74 => usb.on("attach", attachDetected)

### node-usb
- usb.js:515 => usb._enableHotplugEvents()
- src/node_usb.cc:179 => hotplugThis.Reset(info.This().As<Napi::Object>(), 1);

If we disable the line src/node_usb.cc:179 we dont have a crash

## Test by yourself with different Electron versions
Parallelly I created a simple electron project that only loads node-usb and it crashes using electron 9-13 but not in electron 8

`[https://github.com/danielmain/electron-crash-using-usb](https://github.com/danielmain/electron-crash-using-usb)`

I created two bugs on Electron and in node-usb repos respectively

https://github.com/electron/electron/issues/30347
https://github.com/tessel/node-usb/issues/438

We can wait until we get a feedback or we create our own node-usb version, disable that line and re-tests all HW scenarios

@nikolaglumac @tomislavhoracek

**Comment by gabriela-ponce** [comment]
@danielmain I checked the pending issues on build 18857. Here are the results:

- The error in BigSur no longer reproduces after closing Daedalus, nice job 🎉 
- The "Blank screen fix" issue is ok now, but the state is not persisted after closing Daedalus. To clarify: In this build If you choose "Blank screen fix", when you close and reopen Daedalus, it is off again. In the version from production, it's a state that doesn't change until you deactivate it.

File: source/main/environment.js
Comment (nikolaglumac) [inline source/main/environment.js:50]: @danielmain please revert this.

File: source/main/utils/safeExitWithCode.js
Comment (nikolaglumac) [inline source/main/utils/safeExitWithCode.js:13]: @danielmain please revert this.

**Comment by gabriela-ponce** [comment]
@danielmain Here is a list of new issues on the latest build. I'll update the issue summary, but most of the previous issues no longer reproduce:

**macOS:**
- The error "transaction confirmation failed" is displayed if you create a transaction for Ledger S and connect the device after. Check [this](https://drive.google.com/file/d/1AY-sUIDrzKap88KuPp78N2oquTNdJJtY/view?usp=sharing) screenshot.

**Windows:**
- Pair a Ledger S device, wait for it to sync and unlock it. Create a transaction without connecting the device, you'll see that Daedalus asks to "Run cardano app" instead of "Connect the Ledger S". A lot of errors are visible in the console. Check [this](https://drive.google.com/file/d/13ENOwBgwAofb82fq-4y42YQoAsICve9O/view?usp=sharing) screenshot.
Note: If you connect the device at this stage, you still can confirm the transaction.

- [This issue is intermittent, I couldn't get a proper video yet]. When trying to delete a Ledger S wallet, an error on the console is displayed. The process of deletion can take a while and if you restart Daedalus, the wallet is still there. Check [this](https://drive.google.com/file/d/1-Z2WYHAMrb44xRw_WNr4mggGvIn4ghLp/view?usp=sharing) screenshot to see the error.

**All:**
- I was able to reproduce the same issue that Mior mentioned [here](https://github.com/input-output-hk/daedalus/pull/2206#issuecomment-906480448). If you try to move all the funds from a Shelley wallet, you'll get an error message that says there are no enough ADA to cover the fees. Check these screenshots: [transaction](https://drive.google.com/file/d/13ENOwBgwAofb82fq-4y42YQoAsICve9O/view?usp=sharing), [error](https://drive.google.com/file/d/1y76cO5NADvXxzzCi3LSGrJ6dHAn2wwN2/view?usp=sharing). It does not happen in all the Shelley wallets, I think it can be related to wallets that have rewards.

**Comment by nikolaglumac** [comment]
@miorsufianiohk @gabriela-ponce @ManusMcCole @danielmain when can we expect this one to be approved?

**Comment by gabriela-ponce** [comment]
@nikolaglumac So far [the regression is ok.](https://docs.google.com/spreadsheets/d/1ZEvSxJWwwNQA5Grqs_b6Wh3qkt0sfijGNsOwYQSFHfI/edit?usp=sharing) I found a couple of issues that still reproduce from previous builds on macOS:

- If you connect the device after opening the confirmation dialog, you get the message "An error occurred". Check [screenshot](https://drive.google.com/file/d/14LzELWuwnfl8_lUA_Va6qJNaE56KSoMi/view?usp=sharing).

- Can't confirm 2 transactions in a row without disconnecting the device. On the second transaction, the device won't be detected. Check [screenshot](https://drive.google.com/file/d/1gEmoSXxtDki981e2zzRzBfBUWMT7Hq4R/view?usp=sharing).

@miorsufianiohk can you please confirm if you reproduce this in Ledger Nano X?

**Comment by nikolaglumac** [comment]
@gabriela-ponce @miorsufianiohk @ManusMcCole @dmitrii-gaico please re-test all HW scenarios on all platforms and lets get some numbers 🙏

**Comment by miorsufianiohk** [comment]
> @miorsufianiohk can you please confirm if you reproduce this in Ledger Nano X?
@gabriela-ponce , I could confirm that for Nano X on MacOS this happened -> If you connect the device after opening the confirmation dialog, you get the message "An error occurred". [See screenshot](https://drive.google.com/file/d/1rHAP2uh_Lf2njWwRduzHCntHPtPZg7gX/view?usp=sharing).

**Comment by gabriela-ponce** [comment]
@nikolaglumac @danielmain Just finished checking the[ Ledger S cases](https://docs.google.com/spreadsheets/d/1ZEvSxJWwwNQA5Grqs_b6Wh3qkt0sfijGNsOwYQSFHfI/edit?usp=sharing) on build 19005 for all platforms, and these are the findings:

- Windows and Linux are ok.
- [These](https://github.com/input-output-hk/daedalus/pull/2206#issuecomment-913862570) issues still reproduce in macOS. The second issue doesn't happen all the time.

**Comment by nikolaglumac** [comment]
> @nikolaglumac @danielmain Just finished checking the[ Ledger S cases](https://docs.google.com/spreadsheets/d/1ZEvSxJWwwNQA5Grqs_b6Wh3qkt0sfijGNsOwYQSFHfI/edit?usp=sharing) on build 19005 for all platforms, and these are the findings:
> 
> * Windows and Linux are ok.
> * [These](https://github.com/input-output-hk/daedalus/pull/2206#issuecomment-913862570) issues still reproduce in macOS (BigSur). The second issue doesn't happen all the time.

@gabriela-ponce is the same issue reproducible on develop or latest release (4.3.1)?

**Comment by miorsufianiohk** [comment]
> @gabriela-ponce is the same issue reproducible on develop or latest release (4.3.1)?
I am unable to reproduce Gabriela's findings for Nano S on both latest release 4.3.1 and latest develop 19015. Currently testing latest build for this PR. Will report later 👍

**Comment by dmitrii-gaico** [comment]
> > @nikolaglumac @danielmain Just finished checking the[ Ledger S cases](https://docs.google.com/spreadsheets/d/1ZEvSxJWwwNQA5Grqs_b6Wh3qkt0sfijGNsOwYQSFHfI/edit?usp=sharing) on build 19005 for all platforms, and these are the findings:
> > 
> > * Windows and Linux are ok.
> > * [These](https://github.com/input-output-hk/daedalus/pull/2206#issuecomment-913862570) issues still reproduce in macOS (BigSur). The second issue doesn't happen all the time.
> 
> @gabriela-ponce is the same issue reproducible on develop or latest release (4.3.1)?

We have validated both scenarios at MacOS Catalina/BigSur | 4.3.1 current release and latest develop. In all cases issues are NOT reproducible.

**Comment by nikolaglumac** [comment]
Perfect 👌 
Please approve the PR once ready and I will merge it 🚀

**Review by nikolaglumac — APPROVED** [review]
A true masterpiece 💯 
Amazing work @danielmain 🎉

**Review by miorsufianiohk — APPROVED** [review]
LGTM. Tested on 19020 Testnet with Nano S on Win, BigSur, Catalina and Linux 👍

**Review by dmitrii-gaico — APPROVED** [review]

**Pull request merged**
